### DISCUSSION TITLE
Getting started: CLI operations vs options and flags

### DISCUSSION THREAD

**Discussion by Bouke** [gitext-rs/git-stack#110] [Ideas]
The current cli design confuses me. There's a bunch of operations (`--pull`, `--push`, `--rebase`), and separately there are options and flags (`--dry-run`, `--verbose`, `--base` etc). I think it would be great to separate operations from options and flags and drop the `--` prefix from operations. So instead of `git stack --rebase --base master` you would write `git stack rebase --base master`.

**Comment by epage** [comment]
One reason they are options, instead of subcommands, is that you can actually mix them together
```
git stack --pull --fixup squash
```

Another is that `--fixup` takes an argument.  I guess `git stack fixup squash` could work but it'd then be inconsistent with `git stack --rebase --fixup squash` when overriding `stack-auto-fixup`.

Something that will help is we'll be able to separate arguments into distinct groups with the next major version of the CLI argument parsing library we are using (clap)
